internal/builders/docker/pkg/builder.go

```diff
@@ -69,15 +69,15 @@ type Builder struct {
 
 // NewBuilderWithGitFetcher creates a new Builder that fetches the sources
 // from a Git repository.
-func NewBuilderWithGitFetcher(config DockerBuildConfig, forceCheckout bool) (*Builder, error) {
-	gc, err := newGitClient(&config, forceCheckout, 0 /* depth */)
+func NewBuilderWithGitFetcher(config *DockerBuildConfig, forceCheckout bool) (*Builder, error) {
+	gc, err := newGitClient(config, forceCheckout, 0 /* depth */)
 	if err != nil {
 		return nil, fmt.Errorf("could not create builder: %v", err)
 	}
 
 	return &Builder{
 		repoFetcher: gc,
-		config:      config,
+		config:      *config,
```

Comment:
Should the `Builder.config` be a pointer too?

Reply:
`gocritic` suggested to pass by pointer to copy less data. I think `Builder.config` is fine as value so that `Builder` can own its configuration? I think you can judge this better than I can.